### PR TITLE
Adds pets datasets

### LINKED ISSUE #118: Pets dataset
## 🚀 Feature
Related to #116 and #117 , the [pets dataset](http://www.robots.ox.ac.uk/~vgg/data/pets/) also uses VOC annotations

Once the VOC parser lands this should be really fast to implement

### PR DESCRIPTION
**Pull request by lgvaz**
closes #118 

Also introduces the new datasets abstraction discussed in #122

### CODE REVIEW

**Comment by lgvaz** [comment]
@oke-aditya , @ai-fast-track 

I created a [colab notebook](https://colab.research.google.com/drive/1tEwfvy2RqAvD-iIGSgiji-PTlCEkX0ps?usp=sharing) that displays most of the changes done in this PR, please take a look and share your thoughts.

**Comment by lgvaz** [comment]
@ai-fast-track 

This PR still does not implement the changes we discussed today. I'll implement that in a future PR.

File: mantisshrimp/datasets/pets/pets_data.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+CATEGORIES = sorted({"dog", "cat"})
+
+
+def load():
+    save_dir = get_data_dir() / "pets"
+    save_dir.mkdir(exist_ok=True)
+
+    url_images = "http://www.robots.ox.ac.uk/~vgg/data/pets/data/images.tar.gz"
+    images_tar_file = save_dir / "pets.tar.gz"
+    if not images_tar_file.exists():
```
Comment (ai-fast-track) [inline mantisshrimp/datasets/pets/pets_data.py]: Maybe refactor those lines in a function:
```
if not annotations_tar_file.exists():
        download_url(url=url_annotations, save_path=annotations_tar_file)
        shutil.unpack_archive(annotations_tar_file, save_dir)
```

File: mantisshrimp/datasets/pets/pets_data.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        download_url(url=url_images, save_path=images_tar_file)
+        shutil.unpack_archive(images_tar_file, save_dir)
+
+    url_annotations = (
+        "http://www.robots.ox.ac.uk/~vgg/data/pets/data/annotations.tar.gz"
+    )
+    annotations_tar_file = save_dir / "annotations.tar.gz"
+    if not annotations_tar_file.exists():
```
Comment (ai-fast-track) [inline mantisshrimp/datasets/pets/pets_data.py]: Related to previous comment: DRY

```
if not annotations_tar_file.exists():
        download_url(url=url_annotations, save_path=annotations_tar_file)
        shutil.unpack_archive(annotations_tar_file, save_dir)
```

File: examples/fasterrcnn_pets.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+valid_ds = Dataset(valid_records, tfm=valid_tfms)
+
+sample = train_ds[0]
+show_annotation(
+    im=sample["img"], labels=sample["label"], bboxes=sample["bbox"], show=True
+)
+
+# TODO: Rethink CATEGORIES, should they always be present in parser?
+# Should call it CLASSES instead of CATEGORIES?
+model = MantisFasterRCNN(num_classes=len(pets.CATEGORIES))
```
Comment (ai-fast-track) [inline examples/fasterrcnn_pets.py:33]: Not sure about this but: the argument is called `num_classes` so maybe `pets.CATEGORIES` should also be `pets.CLASSES`

File: mantisshrimp/datasets/pets/pets_data.py (hunk 1 above)
Comment (lgvaz) [reply mantisshrimp/datasets/pets/pets_data.py]: Refactored to a function `download_and_extract`

File: examples/fasterrcnn_pets.py (hunk 3 above)
Comment (lgvaz) [reply examples/fasterrcnn_pets.py:33]: This is what I was thinking as well.

Is just that fastai always calls them categories, so I got used to it (and it seems a more descriptive name as well)

But torch always calls them classes...

**Comment by lgvaz** [comment]
After the changes, I'm encountering the following error:

Screenshot 2020-06-30 at 11 46 19 AM

Anyone has any clues? Full strack trace is in the notebook

**Pull request merged**